xml_schema/tests/choice.rs

```diff
@@ -0,0 +1,101 @@
+extern crate yaserde_derive;
+
+use xml_schema_derive::XmlSchema;
+use yaserde::de::from_str;
+use yaserde::ser::to_string;
+
+#[test]
+fn choice() {
+  #[derive(Debug, XmlSchema)]
+  #[xml_schema(source = "xml_schema/tests/choice.xsd")]
+  struct ChoiceTypeSchema;
+
+  use choice_type_schema::xml_schema_types::*;
+
+  let xml_1 = r#"<?xml version="1.0" encoding="UTF-8"?>
+  <person>
+    <firstname>John</firstname>
+  </person>
+  "#;
+
+  let sample_1: Person = from_str(xml_1).unwrap();
+
+  let model = Person {
+    firstname: Some(Firstname {
+      base: "John".to_string(),
+      scope: None,
+    }),
+    lastname: None,
+  };
+
+  assert_eq!(sample_1, model);
+
+  let data = to_string(&model).unwrap();
+  assert_eq!(
+    data,
+    r#"<?xml version="1.0" encoding="utf-8"?><Person><firstname>John</firstname></Person>"#
+  );
+}
+
+#[test]
+fn choice_sequence() {
+  #[derive(Debug, XmlSchema)]
+  #[xml_schema(source = "xml_schema/tests/choice_sequence.xsd")]
+  struct ChoiceTypeSchema;
+
+  use choice_type_schema::xml_schema_types::*;
+
+  let xml_1 = r#"<?xml version="1.0" encoding="UTF-8"?>
+  <person>
+    <name>Doe</name>
+    <firstname>John</firstname>
+  </person>
+  "#;
+
+  let sample_1: Person = from_str(xml_1).unwrap();
+
+  let model = Person {
+    name: "Doe".to_string(),
+    firstname: Some(Firstname {
+      base: "John".to_string(),
+      scope: None,
+    }),
+    lastname: None,
+  };
+
+  assert_eq!(sample_1, model);
+
+  let data = to_string(&model).unwrap();
+  assert_eq!(
+    data,
+    r#"<?xml version="1.0" encoding="utf-8"?><Person><name>Doe</name><firstname>John</firstname></Person>"#
+  );
+}
+
+#[test]
+fn choice_multiple() {
+  #[derive(Debug, XmlSchema)]
+  #[xml_schema(source = "xml_schema/tests/choice_multiple.xsd")]
+  struct ChoiceTypeSchema;
+
+  let xml_1 = r#"<?xml version="1.0" encoding="UTF-8"?>
+  <person>
+    <firstname>John</firstname>
+  </person>
+  "#;
+
+  let sample_1: Person = from_str(xml_1).unwrap();
+
+  let model = Person {
+    firstname: "John".to_string(),
+    lastname: "".to_string(),
+  };
```

Comment:
in that case I expect to have:
```rust
Person {
  parents: vec![Parent::Firstname("John".to_string())],
}


```

Reply:
Yup exactly, I just patched this in one commit just to get it to compile. I am currently working on making this a vector.

Reply:
Oh sorry I just implemented:

```
  let model = Person {
    firstname_list: vec!["John".to_string()],
    lastname_list: vec![],
  };
```

which i notice now is a bit different from what you imagined.

Reply:
I have some attempts which try to implement this as enum, you can see the last commit here: https://github.com/matzipan/xml-schema/pull/new/rebase_impl_choice_enum_attempt

Unforuntately the test fails with:

```
thread 'choice_multiple' panicked at xml_schema/tests/choice.rs:99:3:
assertion `left == right` failed
  left: Gift { content: [] }
 right: Gift { content: [price_history(1), price_history(3)] }
```

So I probably got the yaserde config wrong :(

I will pause this work for now.